AKV 2.0.0 release notes (#954)
dotnet/SqlClient: release-notes/add-ons/AzureKeyVaultProvider/2.0/2.0.0.md

# Release Notes

## General Availability of Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider
_**2.0.0 released 03 March 2021**_

This library contains the implementation of `Microsoft.Data.SqlClient.SqlColumnEncryptionKeyStoreProvider` for accessing Azure Key Vault, and the provider class is named `SqlColumnEncryptionAzureKeyVaultProvider`.

### Added

> - Upgraded AKV Provider to use new Azure Key Vault libraries [#930](https://github.com/dotnet/SqlClient/pull/630)

Reviewer: Shouldn't use "AKV" in public docs, not sure if it applies to release notes.

Reviewer 2: @johnnypham is correct.

> - Upgraded AKV Provider to use new Azure Key Vault libraries [#930](https://github.com/dotnet/SqlClient/pull/630)

Reviewer: ```suggestion
- Upgraded AKV Provider to use new Azure Key Vault libraries [#630](https://github.com/dotnet/SqlClient/pull/630)
```

Author: Could you also include abbreviation removal here in suggestion - let's fix that too.

### Breaking Changes

> - Drops support for .NET Framework 4.6. The new minimum supported .NET Framework version is v4.6.1 [#930](https://github.com/dotnet/SqlClient/pull/630)

Reviewer: ```suggestion
- Drops support for .NET Framework 4.6. The new minimum supported .NET Framework version is v4.6.1 [#630](https://github.com/dotnet/SqlClient/pull/630)
```

This should be #630 too?

> - Updated dependency Mcrosoft.Data.SqlClient on .NET Framework and .NET Core to LTS stable version v1.1.3+ [#946](https://github.com/dotnet/SqlClient/pull/946)

Reviewer 2: ```suggestion
- Updated dependency of Microsoft.Data.SqlClient on .NET Framework and .NET Core to LTS stable version v1.1.3+ [#946](https://github.com/dotnet/SqlClient/pull/946)
```

### Working with SQLColumnEncryptionAzureKeyVaultProvider

> `SqlColumnEncryptionAzureKeyVaultProvider` is implemented against `Microsoft.Data.SqlClient` and supports .NET Framework 4.6.1+, .NET Core 2.1+, and .NET Standard 2.0+. The provider name identifier for this library is "**AZURE_KEY_VAULT**" and it is not registered in the driver by default. Client applications may call the `SqlConnection.RegisterColumnEncryptionKeyStoreProviders()` API once in the lifetime of the driver to register this custom provider by implementing a custom Authentication Callback mechanism.

Reviewer: New provider doesn't use callbacks.

Author: Fixed

Once the provider is registered, it can be used to perform Always Encrypted operations by creating a Column Master Key using the Azure Key Vault Key Identifier URL.

> The linked C# samples below demonstrate using Always Encrypted and Secure Enclaves with Azure Key Vault:

Reviewer 2: I missed this in my first pass. "secure enclaves" should be lower case.
```suggestion
The linked C# samples below demonstrate using Always Encrypted with secure enclaves with Azure Key Vault:
```

> Below linked C# samples demonstrate using Always Encrypted and Secure Enclaves with Azure Key Vault:

Reviewer 2: ```suggestion
The linked C# samples below demonstrate using Always Encrypted and Secure Enclaves with Azure Key Vault:
```

- Legacy API support (Always Encrypted): [AzureKeyVaultProviderExample.cs](https://github.com/dotnet/SqlClient/blob/master/doc/samples/AzureKeyVaultProviderLegacyExample_2_0.cs)
- New API support (Always Encrypted): [AzureKeyVaultProviderExample.cs](https://github.com/dotnet/SqlClient/blob/master/doc/samples/AzureKeyVaultProviderExample_2_0.cs)

> - Legacy API support (Always Encrypted with Secure Enclaves): [AzureKeyVaultProviderExample.cs](https://github.com/dotnet/SqlClient/blob/master/doc/samples/doc\samples\AzureKeyVaultProviderWithEnclaveProviderExample.cs)

Reviewer 2: ```suggestion
- Legacy API support (Always Encrypted with secure enclaves): [AzureKeyVaultProviderExample.cs](https://github.com/dotnet/SqlClient/blob/master/doc/samples/doc\samples\AzureKeyVaultProviderWithEnclaveProviderExample.cs)
```

> - New API support (Always Encrypted with Secure Enclaves): [AzureKeyVaultProviderExample.cs](https://github.com/dotnet/SqlClient/blob/master/doc/samples/doc\samples\AzureKeyVaultProviderWithEnclaveProviderExample_2_0.cs)

Reviewer 2: ```suggestion
- New API support (Always Encrypted with secure snclaves): [AzureKeyVaultProviderExample.cs](https://github.com/dotnet/SqlClient/blob/master/doc/samples/doc\samples\AzureKeyVaultProviderWithEnclaveProviderExample_2_0.cs)
```
